### PR TITLE
osd/OSDMonitor: check svc is writeable before changing pending

### PR DESCRIPTION
**Pull request by batrick**
Fixes: https://tracker.ceph.com/issues/59813

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by ifed01** [comment]
@batrick - looks like make check failures are relevant.

**Comment by batrick** [comment]
> @batrick - looks like make check failures are relevant.

Please have another look.

**Review by ifed01 — APPROVED** [review]

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
ping @rzarzynski

**Review by rzarzynski — APPROVED** [review]
LGTM. Just a nit / question.

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -9814,10 +9815,11 @@ int OSDMonitor::prepare_command_osd_destroy(
   EntityName cephx_entity, lockbox_entity;
   bool idempotent_auth = false, idempotent_cks = false;
 
-  int err = mon.authmon()->validate_osd_destroy(id, uuid,
-                                                 cephx_entity,
-                                                 lockbox_entity,
-                                                 ss);
+  auto&& authmon = mon.authmon();
```
[the remark this answers is not in the document]
Comment (batrick) [reply src/mon/OSDMonitor.cc:9818]: It's just an attempt to make the code resilient to `mon.authmon()` returning a smart pointer someday.

[the remark this answers is not in the document]
Comment (batrick) [reply src/mon/OSDMonitor.cc:9818]: > Do we need to define as an r-value reference? Isn't the usually seen l-value (`auto&`) enough and less surprising?

Also, you can't use `auto&` for an rvalue return (in this case, a raw pointer).

**Comment by ljflores** [comment]
Rados approved: https://tracker.ceph.com/projects/rados/wiki/MAIN#httpstrellocomcPuCOnhYL1841-wip-yuri5-testing-2023-10-02-1105-old-wip-yuri5-testing-2023-09-27-0959

**Pull request merged**